core/src/main/java/io/debezium/connector/cassandra/FileOffsetWriter.java

```diff
@@ -54,11 +58,20 @@ public class FileOffsetWriter implements OffsetWriter {
     private final FileLock snapshotOffsetFileLock;
     private final FileLock commitLogOffsetFileLock;
 
-    public FileOffsetWriter(String offsetDir) throws IOException {
+    private final OffsetFlushPolicy offsetFlushPolicy;
+    private final ExecutorService executorService;
+
+    private long timeOfLastFlush;
+    private long unflushedRecordCount;
+
+    public FileOffsetWriter(String offsetDir, OffsetFlushPolicy offsetFlushPolicy) throws IOException {
         if (offsetDir == null) {
             throw new CassandraConnectorConfigException("Offset file directory must be configured at the start");
         }
 
+        this.offsetFlushPolicy = offsetFlushPolicy;
+        this.timeOfLastFlush = System.currentTimeMillis();
```

Comment:
Please use `io.debezium.util.Clock` or we also have `io.debezium.util.ElapsedTimeStrategy` and similar classes.

Reply:
sounds good. I refactor and move the logic to OffsetFlushPolicy.PeriodicFlushOffsetPolicy class and use io.debezium.util.ElapsedTimeStrategy class.